### PR TITLE
Create reusable functions

### PR DESCRIPTION
**Pull request by jonathanstowell**

### CODE REVIEW

File: src/ZendeskApi.Client/Resources/AbstractBaseResource.cs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            if (scope == null)
+                scope = $"DeleteAsync{idsAsCsv}";
+
+            using (LoggerScope(Logger, scope))
+            using (var client = ApiClient.CreateClient())
+            {
+                var response = await client.DeleteAsync($"{resource}?ids={idsAsCsv}")
+                    .ConfigureAwait(false);
+
+                if (response.StatusCode != HttpStatusCode.OK)
```
Comment (LoadData) [inline src/ZendeskApi.Client/Resources/AbstractBaseResource.cs]: Can we not use `ThrowIfUnsuccessful` instead?

File: src/ZendeskApi.Client/Resources/AbstractBaseResource.cs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        {
+            ApiClient = apiClient;
+            Logger = logger;
+            DocsResource = docsResource;
+        }
+
+        protected async Task<T> GetAsync<T>(
+            string resource,
+            string docs,
+            string scope,
```
Comment (LoadData) [inline src/ZendeskApi.Client/Resources/AbstractBaseResource.cs]: Can we provide a default scope here? As we have done in the `CreateAsync` method?

Comment (jonathanstowell) [reply src/ZendeskApi.Client/Resources/AbstractBaseResource.cs]: In the three examples I looked at the scope was different for almost every usage so it didn't make sense to provide a default here.

Comment (jonathanstowell) [reply src/ZendeskApi.Client/Resources/AbstractBaseResource.cs]: The should be three resource examples but each resource calls this base method multiple times.

**Pull request merged**